pkg/controller/job/job_controller.go

```diff
@@ -153,7 +153,10 @@ func (jm *JobManager) getPodJob(pod *api.Pod) *experimental.Job {
 		glog.V(4).Infof("No jobs found for pod %v, job manager will avoid syncing", pod.Name)
 		return nil
 	}
-	// TODO: add sorting and rethink the overlaping controllers, internally and with RCs
+	if len(jobs) > 1 {
+		glog.Errorf("user error! more than one job is selecting pods with labels: %+v", pod.Labels)
```

Comment:
An event on the job?  @saad-ali can help figure how to make sure the events are coalesced client side to avoid spamming the apiserver.